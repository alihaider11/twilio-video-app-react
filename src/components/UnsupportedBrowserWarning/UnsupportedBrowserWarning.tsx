import React from 'react';
import Video from 'twilio-video';
import { Container, Link, Typography, Paper, Grid } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  container: {
    marginTop: '2.5em',
  },
  paper: {
    padding: '1em',
  },
  heading: {
    marginBottom: '0.4em',
  },
});

export default function({ children }: { children: React.ReactElement }) {
  const classes = useStyles();

  if (!Video.isSupported) {
    return (
      <Container>
        <Grid container justify="center" className={classes.container}>
          <Grid item xs={12} sm={6}>
            <Paper className={classes.paper}>
              <Typography variant="h4" className={classes.heading}>
                Browser not supported
              </Typography>
              <Typography>
                This is not a supported browser. Please refer to our list of{' '}
                <Link
                  href="https://www.twilio.com/docs/video/javascript#supported-browsers"
                  target="_blank"
                  rel="noopener"
                >
                  supported browsers
                </Link>
                .
                <br />
                {!window.isSecureContext && (
                  <>
                    If you are using a browser on the supported browser list, please ensure that this app is served over
                    a secure context. This app must be served over a secure context (e.g. https or localhost). Please
                    see:{' '}
                    <Link
                      href="https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts"
                      target="_blank"
                      rel="noopener"
                    >
                      https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts
                    </Link>
                    .
                  </>
                )}
              </Typography>
            </Paper>
          </Grid>
        </Grid>
      </Container>
    );
  }

  return children;
}
