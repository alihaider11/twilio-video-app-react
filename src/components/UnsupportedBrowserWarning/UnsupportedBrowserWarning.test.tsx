import React from 'react';
import UnsupportedBrowserWarning from './UnsupportedBrowserWarning';
import Video from 'twilio-video';
import { shallow } from 'enzyme';

describe('the UnsupportedBrowserWarning component', () => {
  it('should render correctly when isSupported is false and isSecureContext is true', () => {
    // @ts-ignore
    Video.isSupported = false;
    // @ts-ignore
    window.isSecureContext = true;
    const wrapper = shallow(
      <UnsupportedBrowserWarning>
        <span>Is supported</span>
      </UnsupportedBrowserWarning>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when isSupported is false and isSecureContext is true', () => {
    // @ts-ignore
    Video.isSupported = false;
    // @ts-ignore
    window.isSecureContext = false;
    const wrapper = shallow(
      <UnsupportedBrowserWarning>
        <span>Is supported</span>
      </UnsupportedBrowserWarning>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render children when browser is supported', () => {
    // @ts-ignore
    Video.isSupported = true;
    // @ts-ignore
    window.isSecureContext = true;
    const wrapper = shallow(
      <UnsupportedBrowserWarning>
        <span>Is supported</span>
      </UnsupportedBrowserWarning>
    );
    expect(wrapper.text()).toBe('Is supported');
  });
});
