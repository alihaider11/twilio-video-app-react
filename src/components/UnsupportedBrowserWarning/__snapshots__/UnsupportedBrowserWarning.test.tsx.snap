// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the UnsupportedBrowserWarning component should render correctly when isSupported is false and isSecureContext is true 1`] = `
<WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-container-1"
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2"
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-heading-3"
          variant="h4"
        >
          Browser not supported
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          This is not a supported browser. Please refer to our list of
           
          <WithStyles(ForwardRef(Link))
            href="https://www.twilio.com/docs/video/javascript#supported-browsers"
            rel="noopener"
            target="_blank"
          >
            supported browsers
          </WithStyles(ForwardRef(Link))>
          .
          <br />
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;

exports[`the UnsupportedBrowserWarning component should render correctly when isSupported is false and isSecureContext is true 2`] = `
<WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-container-1"
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2"
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-heading-3"
          variant="h4"
        >
          Browser not supported
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          This is not a supported browser. Please refer to our list of
           
          <WithStyles(ForwardRef(Link))
            href="https://www.twilio.com/docs/video/javascript#supported-browsers"
            rel="noopener"
            target="_blank"
          >
            supported browsers
          </WithStyles(ForwardRef(Link))>
          .
          <br />
          If you are using a browser on the supported browser list, please ensure that this app is served over a secure context. This app must be served over a secure context (e.g. https or localhost). Please see:
           
          <WithStyles(ForwardRef(Link))
            href="https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts"
            rel="noopener"
            target="_blank"
          >
            https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts
          </WithStyles(ForwardRef(Link))>
          .
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
